src/collections/componentSet.ts

```diff
@@ -547,6 +547,13 @@ export class ComponentSet extends LazyCollection<MetadataComponent> {
       return true;
     }
 
+    // This will decode the key of the component before comparing, which can solve some edge cases
+    // in component fullNames such as Layouts. See: https://github.com/forcedotcom/cli/issues/1683
+    const isDirectlyInSetDecoded = this.components.has(decodeURI(simpleKey(component)));
+    if (isDirectlyInSetDecoded) {
+      return true;
+    }
```

Comment:
were you originally doing something else with `isDirectlyInSetDecoded` and this is all that's left?
```suggestion
    if (this.components.has(decodeURI(simpleKey(component)))) {
      return true;
    }
```

Reply:
I was (lazily) following the pattern from above.  I saw this after I put the PR up and forgot to go back and change it.  I'll change it in both spots.